core/network/src/sync.rs

```diff
@@ -452,6 +460,43 @@ impl<B: BlockT> ChainSync<B> {
 		}
 	}
 
+	/// Returns the new sync state of the peer and the block number that 
+	/// we need to request in case the sync state is other than Available.
+	fn handle_ancestor_search_state(
+		n: NumberFor<B>,
+		state: AncestorSearchState<B>,
+		block_hash_match: bool,
+	) -> (PeerSyncState<B>, NumberFor<B>) {
+		match state {
+			AncestorSearchState::LinearSearch(m) => {
+				if block_hash_match {
+					return (PeerSyncState::Available, As::sa(0));
+				}
+				if m < 7 { // 7 is max number of linear search steps (choosen heuristically).
```

Comment:
Could you put this `7` into a constant that is declared outside of the function?

Reply:
Sure (I didn't want to make the function impure and it's not used anywhere else). 
I wonder if 7 is a good number though.

Reply:
Maybe we should test this. In my view that does not make it "impure". But "random" constants in code are not that nice, in my opinion.

Reply:
Agree.